app/src/main/java/com/parishod/watomatic/fragment/ContactSelectorFragment.kt

```diff
@@ -87,19 +87,25 @@ class ContactSelectorFragment : Fragment() {
         snackbar.show()
     }
 
-    private fun addCustomContactDialog() {
+    private fun addCustomContactDialog(errorMessage: String = "") {
         val builder = MaterialAlertDialogBuilder(requireActivity())
             .setTitle(R.string.add_custom_contact)
 
         val input = EditText(activity).also {
             it.inputType = InputType.TYPE_TEXT_VARIATION_PERSON_NAME
+            if (errorMessage.isNotEmpty()) it.error = errorMessage
         }
 
         builder.setPositiveButton(android.R.string.ok) { _, _ ->
             val name = input.text.toString()
             val adapter = binding.contactList.adapter as ContactListAdapter
-            adapter.addCustomName(name)
-            binding.contactList.scrollToPosition(0)
+            if (name.isBlank()) {
+                addCustomContactDialog(getString(R.string.error_name_cannot_be_blank))
```

Comment:
instead of recreating the dialog how about just setting the error to input like 
`input.error = getString(R.string.error_name_cannot_be_blank)`

ofcourse if we are going with this we need to handle dialog dismiss when clicked ok

Reply:
Thats a good idea but as this PR is anyways better than what we have on main branch, accepting it.

Reply:
> if we are going with this we need to handle dialog dismiss when clicked ok

hmm dialog should not be dismissed when clicking on "ok" we have "cancel" button for that.